### PR TITLE
added staticroute plugin

### PR DESCRIPTION
**Pull request by skoef**
for configuring one or more classless static routes with DHCP option 121

### CODE REVIEW

**Comment by skoef** [comment]
@Natolumin you're absolutely right, didn't look too close to that library. I've rewritten the plugin using that dhcpv4.Route instead!

File: cmds/coredhcp/config.yml.example [hunk 1]
```diff
@@ -156,3 +156,9 @@ server4:
         # * lease duration can be given in any format understood by go's
         # "ParseDuration": https://golang.org/pkg/time/#ParseDuration
         - range: leases.txt 10.10.10.100 10.10.10.200 60s
+
+        # optional static routes could be defined as well
+        # - staticroute: <destination>,<gateway> <destination>,<gateway>
+        # where destination should be in CIDR notation and gateway should be
+        # the IP address of the router through which the destination is reachable
```
Comment (Natolumin) [inline cmds/coredhcp/config.yml.example:163]: Maybe worth adding a mention here that local subnet routes can be specified with a gateway of `0.0.0.0` (from https://datatracker.ietf.org/doc/html/rfc3442#page-4 "Local Subnet Routes")

File: cmds/coredhcp/config.yml.example [hunk 2]
```diff
@@ -156,3 +156,9 @@ server4:
         # * lease duration can be given in any format understood by go's
         # "ParseDuration": https://golang.org/pkg/time/#ParseDuration
         - range: leases.txt 10.10.10.100 10.10.10.200 60s
+
+        # optional static routes could be defined as well
+        # - staticroute: <destination>,<gateway> <destination>,<gateway>
```
Comment (Natolumin) [inline cmds/coredhcp/config.yml.example:161]: nit: make it slightly more obvious that this is one or more entries:
```suggestion
        # - staticroute: <destination>,<gateway> [<destination>,<gateway> ...]
```

File: cmds/coredhcp/config.yml.example [hunk 3]
```diff
@@ -156,3 +156,9 @@ server4:
         # * lease duration can be given in any format understood by go's
         # "ParseDuration": https://golang.org/pkg/time/#ParseDuration
         - range: leases.txt 10.10.10.100 10.10.10.200 60s
+
+        # optional static routes could be defined as well
```
Comment (Natolumin) [inline cmds/coredhcp/config.yml.example:160]: nit: this should follow the same style as other plugin descriptions:
```suggestion
        # staticroute advertises additional routes the client should install in its routing table
```

**Review by Natolumin — APPROVED** [review]
A few notes I forgot earlier on the documentation, but otherwise LGTM

File: cmds/coredhcp/config.yml.example (hunk 1 above)
Comment (skoef) [reply cmds/coredhcp/config.yml.example:164]: I know of this option but I usually find it confusing. I updated the comments and mentioned the RFC.

**Review by Natolumin — APPROVED** [review]

**Pull request merged**